wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsWorkUnitsClient.java

```diff
@@ -378,10 +379,15 @@ static public boolean isWorkunitComplete(WUState state)
      */
     public static WUState getStateID(WorkunitInfo wu)
     {
-        if (wu != null)
-            return getStateID(wu.getStateID());
-        else
-            return WUState.UNKNOWN;
+        if (wu != null) {
+            if (wu.getStateID() != null) {
```

Comment:
@lrenn as part of the Pull Request process you checked off the "My code follows the code style of this project. " but I noticed " I have applied the Eclipse code-format template provided." was not checked off. 
Please take a look at this link: https://github.com/hpcc-systems/HPCC-JAPIs#contributions